src/uri.rs

```diff
@@ -389,6 +389,48 @@ impl StdError for UriError {
     }
 }
 
+// TODO: Is ByteStr adding value? Inline it?
```

Comment:
Storing a `&str` isn't necessary. The API of `ByteStr` works just fine. The point is that we kind of want a `String`, but one that can be shared, since we receive `Bytes` when parsing new requests.